code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -73,18 +73,18 @@
 // Pipenet stuff; housekeeping
 
 /obj/machinery/atmospherics/components/nullifyNode(i)
-	if(parents[i])
-		nullifyPipenet(parents[i])
-		QDEL_NULL(airs[i])
-	..()
+	// Every node has a parent pipeline and an air associated with it.
+	nullifyPipenet(parents[i])
+	QDEL_NULL(airs[i])
+	return ..()
```

Comment:
```suggestion
	// Every node has a parent pipeline and an air associated with it.
	if(parents[i])
		nullifyPipenet(parents[i])
	QDEL_NULL(airs[i])
	return ..()
```

`if(nodes[i])` is the problem, and it means that harddels aren't caused by qdel()ing components with unconnected nodes (but it does leave undeleted pipenets so there's still memory leak to be fixed)